src/spec/datatypes.rs

```diff
@@ -0,0 +1,388 @@
+/*!
+ * Data Types
+*/
+use std::{fmt, ops::Index};
+
+use serde::{
+    de::{Error, IntoDeserializer},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+#[serde(untagged)]
+/// All data types are either primitives or nested types, which are maps, lists, or structs.
+pub enum Type {
+    /// Primitive types
+    Primitive(PrimitiveType),
+    /// Struct type
+    Struct(StructType),
+    /// List type.
+    List(ListType),
+    /// Map type
+    Map(MapType),
+}
+
+impl fmt::Display for Type {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Type::Primitive(primitive) => write!(f, "{}", primitive),
+            Type::Struct(_) => write!(f, "struct"),
+            Type::List(_) => write!(f, "list"),
+            Type::Map(_) => write!(f, "map"),
+        }
+    }
+}
+
+/// Primitive data types
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+#[serde(rename_all = "lowercase", remote = "Self")]
+pub enum PrimitiveType {
+    /// True or False
+    Boolean,
+    /// 32-bit signed integer
+    Int,
+    /// 64-bit signed integer
+    Long,
+    /// 32-bit IEEE 753 floating bit.
+    Float,
+    /// 64-bit IEEE 753 floating bit.
+    Double,
+    /// Fixed point decimal
+    Decimal {
+        /// Precision
+        precision: u32,
+        /// Scale
+        scale: u32,
+    },
+    /// Calendar date without timezone or time.
+    Date,
+    /// Time of day without date or timezone.
+    Time,
+    /// Timestamp without timezone
+    Timestamp,
```

Comment:
Spark uses `Timestamp` to represent timestamp with timezone, `TimestampNtz` to represent timestamp without timezone

Reply:
I would also prefer to stick to the iceberg spec.

Reply:
I find the negation rather confusing, on top of that, we should encourage people to use timestamps without timezones =)